File: video.cpp
// g++ video.cpp `pkg-config opencv --cflags` `pkg-config opencv --libs` -o video.out

#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    char *video_path;

    if(argc==2)
    {
        video_path = argv[1];
    }
    else
    {
        video_path = (char*)"/nas/android_movies/wanfi_2015Feb/shot_from_wanfi/B_WeiLun_D024/DMOV_0006.mp4";
    }

    cv::VideoCapture capture;
    if(capture.open(video_path) == false){
        fprintf(stderr, "can't open video: %s\n", video_path);
        return 0;
    }

    double fps = capture.get(CV_CAP_PROP_FPS);
    capture.release();
    fprintf(stderr, "fps: %.2f\n", fps);

    return 0;
}

File: video.py
import cv2 
import sys 

if len(sys.argv) == 2:
    video_path = sys.argv[1]
else:
    video_path = "/nas/android_movies/wanfi_2015Feb/shot_from_wanfi/B_WeiLun_D024/DMOV_0006.mp4"

cap = cv2.VideoCapture(video_path)
if not cap:
    print "can't open video:", video_path
    sys.exit(0)

print "fps:", cap.get(cv2.cv.CV_CAP_PROP_FPS)
sys.exit(0)

while True:
    ret, frame = cap.read()

    if frame is None:
        print "no frame"
        break

    cv2.imshow('video', frame)
    if cv2.waitKey(1) & 0xff == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()


